File: monitoring-server/reporting-service-api/src/main/scala/com/qwerfah/reporting/resources/ModelStat.scala
package com.qwerfah.reporting.resources

import com.qwerfah.common.Uid

/** Equipment model statistics.
  * @param modelUid
  *   Model identifier.
  * @param modelName
  *   Model name.
  * @param paramCount
  *   Total number of model params.
  * @param instanceCount
  *   Total number of model deployed instances.
  * @param activeInstanceCount
  *   Number of active model instances.
  * @param inactiveInstanceCount
  *   Number of inactive model instances.
  * @param decommissionedInstanceCount
  *   Number of decommissioned model instances.
  * @param monitorCount
  *   Totla number of monitors for all model instances.
  */
final case class ModelStat(
  modelUid: Uid,
  modelName: String,
  paramCount: Int,
  instanceCount: Int,
  activeInstanceCount: Int,
  inactiveInstanceCount: Int,
  decommissionedInstanceCount: Int,
  monitorCount: Int
)

File: monitoring-client/src/app/models/model-stat.ts
export class ModelStat {
  constructor(
    public modelUid: string,
    public modelName: string,
    public paramCount: number,
    public instanceCount: number,
    public activeInstanceCount: number,
    public inactiveInstanceCount: number,
    public decommissionedInstanceCount: number,
    public monitorCount: number
  ) {}
}
